EKF/ekf_helper.cpp

```diff
@@ -1196,18 +1196,22 @@ bool Ekf::reset_imu_bias()
 // Innovation Test Ratios - these are the ratio of the innovation to the acceptance threshold.
 // A value > 1 indicates that the sensor measurement has exceeded the maximum acceptable level and has been rejected by the EKF
 // Where a measurement type is a vector quantity, eg magnetometer, GPS position, etc, the maximum value is returned.
-void Ekf::get_innovation_test_status(uint16_t *status, float *mag, float *vel, float *pos, float *hgt, float *tas, float *hagl, float *beta)
+void Ekf::get_innovation_test_status(uint16_t *status, float *mag, float *gps_vel, float *gps_pos, float *ev_vel, float *ev_pos, float *hgt, float *tas, float *hagl, float *beta)
 {
 	// return the integer bitmask containing the consistency check pass/fail status
 	*status = _innov_check_fail_status.value;
 	// return the largest magnetometer innovation test ratio
 	*mag = sqrtf(math::max(_yaw_test_ratio, math::max(math::max(_mag_test_ratio[0], _mag_test_ratio[1]), _mag_test_ratio[2])));
-	// return the largest NED velocity innovation test ratio
-	*vel = sqrtf(math::max(math::max(_vel_pos_test_ratio[0], _vel_pos_test_ratio[1]), _vel_pos_test_ratio[2]));
-	// return the largest NE position innovation test ratio
-	*pos = sqrtf(math::max(_vel_pos_test_ratio[3], _vel_pos_test_ratio[4]));
+	// return the largest NED GPS velocity innovation test ratio
+	*gps_vel = sqrtf(math::max(math::max(_gps_vel_pos_test_ratio[0], _gps_vel_pos_test_ratio[1]), _gps_vel_pos_test_ratio[2]));
+	// return the largest NE GPS position innovation test ratio
+	*gps_pos = sqrtf(math::max(_gps_vel_pos_test_ratio[3], _gps_vel_pos_test_ratio[4]));
+	// return the largest external vision velocity innovation test ratio
+	*ev_vel = sqrtf(math::max(math::max(_ev_vel_pos_test_ratio[0], _ev_vel_pos_test_ratio[1]), _ev_vel_pos_test_ratio[2]));
+	// return the largest horizontal external vision position innovation test ratio
+	*ev_pos = sqrtf(math::max(_ev_vel_pos_test_ratio[3], _ev_vel_pos_test_ratio[4]));
 	// return the vertical position innovation test ratio
-	*hgt = sqrtf(_vel_pos_test_ratio[5]);
+	*hgt = sqrtf(_gps_vel_pos_test_ratio[5])+sqrtf(_ev_vel_pos_test_ratio[5]);
```

Comment:
I think we should return the maximum.

Reply:
I am still not sure what would be the appropriate solution. Can you give me some reasoning why we should use the maximum?

Reply:
This is resolved in the newest version.